src/sdltiles.cpp

```diff
@@ -2029,6 +2067,37 @@ void load_soundset() {
             playlists[playlist_id] = playlist_to_load;
         }
     }
+
+    // Load sound effects. This loads the sound effect chunks directly
+    // into memory.
+    location = FILENAMES["datadir"] + "/sound/soundset.json";
+    std::ifstream jsonstream2(location.c_str(), std::ifstream::binary);
+    if (jsonstream2.good()) {
+        JsonIn json(jsonstream2);
+        JsonObject config = json.get_object();
+        JsonArray sound_effects_json = config.get_array("sound_effects");
+        for (unsigned i=0; i < sound_effects_json.size(); i++) {
+            JsonObject sound_effect = sound_effects_json.get_object(i);
+
+            std::string sound_effect_id = sound_effect.get_string("id");
+            std::string sound_effect_variant = sound_effect.get_string("variant", "default");
+            sound_effect_entry sfx_to_load;
+
+            const std::string filename = sound_effect.get_string("file");
+            const std::string path = (FILENAMES["datadir"] + "/sound/" + filename);
+            Mix_Chunk *loaded_chunk = Mix_LoadWAV(path.c_str());
```

Comment:
I see a `Mix_LoadWAV`, but I don't see a `Mix_FreeChunk`. Who and when is the chunk freed?

Instead of an ugly raw pointer, you might want to use a `shared_ptr` plus a proper delete function:
```
struct delchunk_functor {
    void operator()( Mix_Chunk *c ) const {
        Mix_FreeChunk( c );
    }
};
typedef std::shared_ptr<Mix_Chunk, delchunk_functor> CHUNK_PTR;
```
Using `CHUNK_PTR` instead of the raw pointer in `sound_effect_entry` will automatically delete the check when the last reference to it is deleted (which happens when the program exists).